plugin/kubernetes/controller.go

```diff
@@ -423,6 +430,9 @@ func (dns *dnsControl) detectChanges(oldObj, newObj interface{}) {
 	switch ob := obj.(type) {
 	case *object.Service:
 		dns.updateModifed()
+		if len(ob.ExternalIPs) > 0 {
```

Comment:
can we check if the actual external IPs have changed?

Reply:
Yeah - we also dont check if other Service attributes have changed... e.g.
* externalName, which would change CNAME records
* defined ports, which would change SRV records

ClusterIP is immutable, so don't need to check that...

Reply:
Added checks in f430772.  I tested manually, with a local cluster.
We don't have good unit test coverage in the controller.